File: src/main/java/tn/esprit/projet/models/PatientPsychologue.java
package tn.esprit.projet.models;

import java.time.LocalDate;

public class PatientPsychologue {
    private int id;
    private int patientId;
    private int psychologueId;
    private LocalDate dateDebut;
    private LocalDate dateFin;
    private String statut;

    // Informations supplémentaires pour l'affichage
    private String patientNom;
    private String patientPrenom;
    private String patientEmail;
    private String psychologueNom;
    private String psychologuePrenom;

    // Constructeurs
    public PatientPsychologue() {}

    public PatientPsychologue(int patientId, int psychologueId) {
        this.patientId = patientId;
        this.psychologueId = psychologueId;
        this.dateDebut = LocalDate.now();
        this.statut = "actif";
    }

    // Getters et Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getPatientId() { return patientId; }
    public void setPatientId(int patientId) { this.patientId = patientId; }

    public int getPsychologueId() { return psychologueId; }
    public void setPsychologueId(int psychologueId) { this.psychologueId = psychologueId; }

    public LocalDate getDateDebut() { return dateDebut; }
    public void setDateDebut(LocalDate dateDebut) { this.dateDebut = dateDebut; }

    public LocalDate getDateFin() { return dateFin; }
    public void setDateFin(LocalDate dateFin) { this.dateFin = dateFin; }

    public String getStatut() { return statut; }
    public void setStatut(String statut) { this.statut = statut; }

    public String getPatientNom() { return patientNom; }
    public void setPatientNom(String patientNom) { this.patientNom = patientNom; }

    public String getPatientPrenom() { return patientPrenom; }
    public void setPatientPrenom(String patientPrenom) { this.patientPrenom = patientPrenom; }

    public String getPatientEmail() { return patientEmail; }
    public void setPatientEmail(String patientEmail) { this.patientEmail = patientEmail; }

    public String getPsychologueNom() { return psychologueNom; }
    public void setPsychologueNom(String psychologueNom) { this.psychologueNom = psychologueNom; }

    public String getPsychologuePrenom() { return psychologuePrenom; }
    public void setPsychologuePrenom(String psychologuePrenom) { this.psychologuePrenom = psychologuePrenom; }

    public String getPatientFullName() {
        return patientPrenom + " " + patientNom;
    }

    public String getPsychologueFullName() {
        return psychologuePrenom + " " + psychologueNom;
    }
}

File: src/Entity/PatientPsychologue.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

use App\Repository\PatientPsychologueRepository;

#[ORM\Entity(repositoryClass: PatientPsychologueRepository::class)]
#[ORM\Table(name: 'patient_psychologue')]
class PatientPsychologue
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(int $id): self
    {
        $this->id = $id;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: false)]
    private ?int $patient_id = null;

    public function getPatient_id(): ?int
    {
        return $this->patient_id;
    }

    public function setPatient_id(int $patient_id): self
    {
        $this->patient_id = $patient_id;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: false)]
    private ?int $psychologue_id = null;

    public function getPsychologue_id(): ?int
    {
        return $this->psychologue_id;
    }

    public function setPsychologue_id(int $psychologue_id): self
    {
        $this->psychologue_id = $psychologue_id;
        return $this;
    }

    #[ORM\Column(type: 'date', nullable: true)]
    private ?\DateTimeInterface $date_debut = null;

    public function getDate_debut(): ?\DateTimeInterface
    {
        return $this->date_debut;
    }

    public function setDate_debut(?\DateTimeInterface $date_debut): self
    {
        $this->date_debut = $date_debut;
        return $this;
    }

    #[ORM\Column(type: 'date', nullable: true)]
    private ?\DateTimeInterface $date_fin = null;

    public function getDate_fin(): ?\DateTimeInterface
    {
        return $this->date_fin;
    }

    public function setDate_fin(?\DateTimeInterface $date_fin): self
    {
        $this->date_fin = $date_fin;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $statut = null;

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(?string $statut): self
    {
        $this->statut = $statut;
        return $this;
    }

}
